src/Akeneo/Connectivity/Connection/back/Infrastructure/Persistence/Elasticsearch/Query/SearchEventSubscriptionDebugLogsQuery.php

```diff
@@ -80,43 +85,74 @@ public function execute(string $connectionCode, ?string $encryptedSearchAfter =
             'size' => 25,
             'sort' => [
                 'timestamp' => 'DESC',
-                '_id' => 'ASC'
+                '_id' => 'ASC',
             ],
             'track_total_hits' => true,
             'query' => [
                 'bool' => [
                     'should' => [
-                        ['bool' => ['must' => [
-                            ['terms' => ['_id' => $lastNoticeAndInfoIds['ids']]],
-                        ]]],
-                        ['bool' => ['must' => [
-                            ['terms' => ['level' => ['error', 'warning']]],
-                            ['range' => ['timestamp' => ['gte' => $nowTimestamp - self::MAX_LIFETIME_OF_WARNING_AND_ERROR_LOGS]]],
-                            ['bool' => ['should' => [
-                                ['term' => ['connection_code' => $connectionCode]],
-                                ['bool' => ['must_not' => ['exists' => ['field' => 'connection_code']]]], // connection_code IS NULL
-                            ]]],
-                        ]]],
+                        [
+                            'bool' => [
+                                'must' => [
+                                    ['terms' => ['_id' => $lastNoticeAndInfoIds['ids']]],
+                                ],
+                            ],
+                        ],
+                        [
+                            'bool' => [
+                                'must' => [
+                                    [
+                                        'terms' => [
+                                            'level' => [
+                                                EventsApiDebugLogLevels::ERROR,
+                                                EventsApiDebugLogLevels::WARNING,
+                                            ],
+                                        ],
+                                    ],
+                                    ['range' => ['timestamp' => ['gte' => $nowTimestamp - self::MAX_LIFETIME_OF_WARNING_AND_ERROR_LOGS]]],
+                                    [
+                                        'bool' => [
+                                            'should' => [
+                                                ['term' => ['connection_code' => $connectionCode]],
+                                                ['bool' => ['must_not' => ['exists' => ['field' => 'connection_code']]]],
+                                                // connection_code IS NULL
+                                            ],
+                                        ],
+                                    ],
+                                ],
+                            ],
+                        ],
                     ],
                 ],
             ],
         ];
 
+        if ($filters['levels']) {
+            $query['query']['bool']['must'] = ['terms' => ['level' => $filters['levels']]];
```

Comment:
```suggestion
            $query['query']['bool']['must'] = ['terms' => ['level' => $filters['levels']]];
```
Why does this line is not formatted ?

Reply:
Done